docs/current/guides/252029-build-local-images.md

```diff
@@ -0,0 +1,207 @@
+  - It uses the previous `Container` object's `WithNewFile()` method to create a new file at the NGINX web server root and return the result as a new `Container`.
+  - It uses the `Export()` method to write the final container image to the host filesystem as a TAR file at `/tmp/my-nginx.tar`.
+
+</TabItem>
+<TabItem value="Node.js">
+
+```javascript file=./snippets/build-local-images/export/index.mjs
+```
+
+This code listing performs the following operations:
+  - It imports the Dagger client library.
+  - It creates a Dagger client with `connect()`. This client provides an interface for executing commands against the Dagger engine.
+  - It uses the client's `container().from()` method to initialize a new container from a base image (`nginx:1.23-alpine`). The additional `platform` argument to the `container()` method instructs Dagger to build for a specific architecture (`linux/amd64`). This method returns a `Container` representing an OCI-compatible container image.
+  - It uses the previous `Container` object's `withNewFile()` method to create a new file at the NGINX web server root and return the result as a new `Container`.
+  - It uses the `export()` method to write the final container image to the host filesystem as a TAR file at `/tmp/my-nginx.tar`.
+
+</TabItem>
+<TabItem value="Python">
+
+```python file=./snippets/build-local-images/export/main.py
+```
+
+This code listing performs the following operations:
+  - It imports the Dagger client library.
+  - It creates a Dagger client with `dagger.Connection()`. This client provides an interface for executing commands against the Dagger engine.
+  - It uses the client's `container().from_()` method to initialize a new container from a base image (`nginx:1.23-alpine`). The additional `platform` argument to the `container()` method instructs Dagger to build for a specific architecture (`linux/amd64`). This method returns a `Container` representing an OCI-compatible container image.
+  - It uses the previous `Container` object's `with_new_file()` method to create a new file at the NGINX web server root and return the result as a new `Container`.
+  - It uses the `export()` method to write the final container image to the host filesystem as a TAR file at `/tmp/my-nginx.tar`.
+
+</TabItem>
+</Tabs>
+
+Once exported, the image can be imported into Docker with `docker load`:
+
+```shell
+docker load -i /tmp/my-nginx.tar
+...
+Loaded image ID: sha256:2e340...
+```
+
+Once imported, the image can be used like any other Docker image. For example, to run the sample image above, use the following `docker run` command:
+
+```shell
+docker run --rm --net=host 2e340...
+```
+
+## Approach 2: Use a local registry server
```

Comment:
@marcosnils expressed some concerns about the DX under this approach. I agree it's not great but at the same time, this is a very common question in Discord (see the parent issue for links to user questions), so it might make sense to document it anyway. Or maybe we can improve it in some way? 

Interested to hear thoughts on above from @vito @marcosnils @sipsma @shykes @samalba

Reply:
Yeah, I think the DX here is frankly pretty terrible, but I agree that it's worth documenting just to have the answer out there.

Maybe we can add a warning or something that this isn't necessarily the recommended practice at the moment, owing to the bad DX, and it's just here for completeness?

Reply:
Added

Reply:
My concern is not so much with the bad DX, but the fact that it's under a pretty wide and generic title. This makes Dagger look bad, because it makes it look like "building an image for local dev on Dagger" is horribly complicated. When really, "building an image" is actually a good experience in Dagger; and it's specifically "loading an image into a local Docker engine" that's complicated. So my suggestion would be to change the title (and perhaps the intro paragraph) to make it clear that this complicated / bad DX is to solve a very narrow / specific problem (exporting images into a local docker engine). And not necessarily the only way to either "build an image" or "do local development".

Reply:
@shykes, thanks for this! Updated per your feedback.